core/src/providers/anthropic.rs

```diff
@@ -176,47 +299,90 @@ impl LLM for AnthropicLLM {
         top_p: Option<f32>,
         top_logprobs: Option<i32>,
         _extras: Option<Value>,
-        _event_sender: Option<UnboundedSender<Value>>,
+        event_sender: Option<UnboundedSender<Value>>,
     ) -> Result<LLMGeneration> {
         assert!(self.api_key.is_some());
         assert!(n > 0);
 
-        // anthropic only supports generating one sample at a time
-        // so we loop here and make n API calls
-        let mut completions: Vec<Tokens> = vec![];
-        for _ in 0..n {
-            let response = self
-                .generate(
-                    prompt,
-                    match max_tokens {
-                        Some(m) => m,
-                        None => 256,
-                    },
-                    temperature,
-                    match top_p {
-                        Some(p) => p,
-                        None => 1.0,
-                    },
-                    match top_logprobs {
-                        Some(k) => k,
-                        None => 0,
-                    },
-                    stop,
-                )
-                .await?;
-
-            completions.push(Tokens {
-                // Anthropic only return the text
-                text: response.completion.clone(),
-                tokens: Some(vec![]),
-                logprobs: Some(vec![]),
-                top_logprobs: Some(vec![]),
-            });
-        }
+        let c: Vec<Tokens> = match event_sender {
+            Some(es) => {
```

Comment:
You ignore `n` here silently. Let's add an error if `n>1`.

Since the logic for `n>1` is pretty broken for non streamed completion as well (sequential which would be terribly slow) let's clean-up that logic as well and just return an informative error if `n>1` for all types of completions with Antropic. Important that it's not an `assert` but an error being returned.

Reply:
I've checked OpenAI behavior with n>1 when streaming. It works well but we don't support it. If you have time lets add an error in the OpenAI `completion` method when `n>1` and `event_sender` is set :pray: